python/python/benchmarks/test_take.py

```diff
@@ -0,0 +1,128 @@
+# SPDX-License-Identifier: Apache-2.0
+# SPDX-FileCopyrightText: Copyright The Lance Authors
+import os
+import tempfile
+import uuid
+
+import lance
+import numpy as np
+import pyarrow as pa
+import pytest
+from lance import LanceDataset
+
+DEFAULT_BATCH_SIZE = 1024
+ENV_OBJECT_STORAGE_TEST_DATASET_URI_PREFIX = (
+    "LANCE_OBJECT_STORAGE_TEST_DATASET_URI_PREFIX"
+)
+
+
+def get_path_prefixes():
+    temp_dir = tempfile.mkdtemp()
+    prefixes = ["memory://", f"file://{temp_dir}"]
+    object_store_path_prefix = os.getenv(ENV_OBJECT_STORAGE_TEST_DATASET_URI_PREFIX, "")
+    if object_store_path_prefix:
+        prefixes.append(object_store_path_prefix)
+    return prefixes
+
+
+def create_dataset(
+    path: str,
+    data_storage_version,
+    num_batches: int,
+    file_size: int,
+    batch_size: int = DEFAULT_BATCH_SIZE,
+) -> LanceDataset:
+    schema = pa.schema(
+        [
+            pa.field("i", pa.int32(), nullable=False),
+            pa.field("f", pa.float32(), nullable=False),
+            pa.field("s", pa.binary(), nullable=False),
+            pa.field("fsl", pa.list_(pa.float32(), 2), nullable=False),
+            pa.field("blob", pa.binary(), nullable=False),
+        ]
+    )
+
+    batches = []
+    for i in range(num_batches):
+        start = i * batch_size
+        stop = (i + 1) * batch_size
+        arr_i = pa.array(np.arange(start, stop, dtype=np.int32))
+        arr_f = pa.array(np.arange(start, stop, dtype=np.float32))
+        arr_s = pa.array([f"blob-{x}".encode() for x in range(start, stop)])
+        fsl_values = np.array(
+            [
+                float(batch_size + (x - batch_size) // 2)
+                for x in range(start, stop + batch_size)
+            ],
+            dtype=np.float32,
+        )
+        arr_fsl = pa.FixedSizeListArray.from_arrays(pa.array(fsl_values), 2)
+        arr_blob = pa.array([f"blob-{x}".encode() for x in range(start, stop)])
+
+        batch = pa.RecordBatch.from_arrays(
+            [arr_i, arr_f, arr_s, arr_fsl, arr_blob], schema=schema
+        )
+        batches.append(batch)
+
+    table = pa.Table.from_batches(batches, schema=schema)
+
+    storage_options = None
+    if not path.startswith(("memory://", "file://")):
+        storage_options = {
+            "access_key_id": os.getenv("ACCESS_KEY_ID", ""),
```

Comment:
IMO, We can let Pylance check these config options. If missed, it would throw an exception.